lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs

```diff
@@ -3163,75 +3162,132 @@ instance Buildable a => Show (ShowBuildable a) where
 instance Arbitrary (Hash "Datum") where
     arbitrary = pure $ Hash $ BS.pack $ replicate 28 0
 
+instance IsCardanoEra era => Arbitrary (Cardano.AddressInEra era) where
+      arbitrary = genAddressInEra Cardano.cardanoEra
+
+instance IsCardanoEra era => Arbitrary (Cardano.TxOutDatum ctx era) where
+      arbitrary = genTxOutDatum Cardano.cardanoEra
+
+instance IsCardanoEra era => Arbitrary (Cardano.TxOut ctx era) where
+      arbitrary = genTxOut Cardano.cardanoEra
+      shrink (Cardano.TxOut addr val dat refScript) = tail
+          [ Cardano.TxOut addr' val' dat' refScript'
+          | addr' <- prependOriginal shrink addr
+          , val' <- prependOriginal shrink val
+          , dat' <- prependOriginal shrink dat
+          , refScript' <- prependOriginal (const []) refScript
+          ]
+
+instance IsCardanoEra era => Arbitrary (Cardano.TxOutValue era) where
+      arbitrary = case Cardano.cardanoEra @era of
+         Cardano.AlonzoEra ->
+             Cardano.TxOutValue Cardano.MultiAssetInAlonzoEra
+                 <$> genValueForTxOut
+         Cardano.BabbageEra ->
+             Cardano.TxOutValue Cardano.MultiAssetInBabbageEra
+                 <$>  genValueForTxOut
+         e -> error $ mconcat
+             [ "Arbitrary (TxOutValue "
+             , show e
+             , ") not yet implemented)"
+             ]
+      shrink (Cardano.TxOutValue Cardano.MultiAssetInAlonzoEra val) =
+          map
+              (Cardano.TxOutValue Cardano.MultiAssetInAlonzoEra
+                  . toCardanoValue)
+              (shrink $ Compatibility.fromCardanoValue val)
```

Comment:
I suspect the GADTs will mess it up here in the case expression, so will refrain from attempting. When switch to using Cardano.Ledger.Mary.Value it would probably work out.